gcp-resources/src/main/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProvider.java

```diff
@@ -33,14 +33,14 @@
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.common.AttributesBuilder;
 import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
-import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
 import io.opentelemetry.sdk.resources.Resource;
 import io.opentelemetry.semconv.ResourceAttributes;
 import java.util.Map;
 import java.util.Optional;
 import java.util.logging.Logger;
 
-public class GCPResourceProvider implements ResourceProvider {
+public class GCPResourceProvider implements ConditionalResourceProvider {
```

Comment:
Any reason for `GCPResourceProvider` to implement `ConditionalResourceProvider` vs extend `CloudResourceProvider`?

Reply:
We'd need to extract `CloudResourceProvider` into a separate project - it seems a bit overkill given how little it (currently) does.

Reply:
Thanks for the explanation

Reply:
I could not find any reference to the `CloudResourceProvider` - could you provide some context on this ?

Reply:
This change looks ok to me. One question to gather more context on the change -
What is the difference between `ConditionalResourceProvider` and `CloudResourceProvider`? They seem similar to me.

Reply:
"conditional" an interface, "cloud" a class that implements "conditional"

https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1225/files#diff-f7f812ee0092f9ddc5fd12dcb198135e3a7d1083147368a0071acc54296d1335R13

Reply:
No, I meant why couldn't the classes in aws resource detector directly implement the interface ?

Reply:
copy pasting the content of cloud resource provider?

Reply:
Ah, I see, mostly to avoid redundant code across multiple classes. Makes sense. Apologies for the confusion.